PlayerSession.py

```diff
@@ -0,0 +1,146 @@
+from ServerConnection import ServerConnection
+from enum import IntEnum
+from Aliases import jsonDict
+
+class Result(IntEnum):
+    """
+    Enumerator of all possible result codes given in the docs.
+    """
+    OKAY = 0
+    BAD_COMMAND = 1
+    ACCESS_DENIED = 2
+    INAPPROPRIATE_GAME_STATE = 3
+    TIMEOUT = 4
+    INTERNAL_SERVER_ERROR = 500
+
+    def __eq__(self, other):
+        if isinstance(other, Result):
+            return self.value == other.value
+
+        return self.value == other  # if it's int
+
+
+class PlayerSession:
+    __slots__ = ("name", "password", "connection")  # class members
+
+    def __init__(self, name, password):
+        self.name = name
+        self.password = password
+
+    def __enter__(self):
+        self.connection = ServerConnection()
+        return self
+
+    @staticmethod
+    def __handleResult(result):
+        """
+        Handles error codes and returns data if request was successful
+        :param result: result dict from action request with "resultCode" and "data" keys
+        :return: result["data"] if the result was okay,
+        None if it was TIMEOUT error.
+        Exception is raised for other error types.
+        """
+        code = result["resultCode"]
+        if code == Result.BAD_COMMAND or code == Result.INAPPROPRIATE_GAME_STATE or code == Result.ACCESS_DENIED or code == Result.INTERNAL_SERVER_ERROR:
```

Comment:
if code in (Result.BAD_COMMAND, INAPPROPRIATE_GAME_STATE ...)
This will look nicer